src/main/kotlin/org/rust/ide/settings/RsPathsExcludeTable.kt

```diff
@@ -0,0 +1,135 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.settings
+
+import com.intellij.execution.util.ListTableWithButtons
+import com.intellij.icons.AllIcons
+import com.intellij.ide.ui.laf.darcula.DarculaUIUtil
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.ComboBoxTableRenderer
+import com.intellij.ui.components.fields.ExtendableTextField
+import com.intellij.ui.scale.JBUIScale
+import com.intellij.util.containers.map2Array
+import com.intellij.util.ui.ColumnInfo
+import com.intellij.util.ui.ListTableModel
+import javax.swing.DefaultCellEditor
+import javax.swing.JTextField
+import javax.swing.table.DefaultTableCellRenderer
+import javax.swing.table.TableCellEditor
+import javax.swing.table.TableCellRenderer
+
+data class Item(var path: String, var type: ExclusionType, var scope: ExclusionScope)
+enum class ExclusionScope { Project, IDE }
```

Comment:
Maybe move them after the `RsPathsExcludeTable` class body or turn them to nested classes?